### ISSUE TITLE
add an option to not automatically generate classifier

### ISSUE THREAD

**Issue by deronnax** [tox-dev/pyproject-fmt#176]
Hello.
I set up pyproject-fmt for some of our projects which are apps, not libraries, not meant to be upload to PyPI, and I find the classifiers automatic generation unwanted for our case. It doesn't make sense for a private app to have PyPI classifiers. I would like an option to turn their generation off.
I can do the PR if you are OK with this.
Cheers.

**Comment by gaborbernat** [comment]
I'm trying to limit the number of knobs this tool has, and this feels one of those that would be of little benefit and expand our API. What's wrong with adding the classifiers? It will not hurt your app 🤔

**Comment by deronnax** [comment]
It could, I think, confuse a reader of the pyproject.toml in a hurry that the project is a package (plus some bit of useless git noise each time the supported python version is changed).

**Comment by gaborbernat** [comment]
Classifiers are not useful only for libraries, they are IMHO valid for applications too. Classifiers are just metadata explaining traits of the app, the way I see it.

**Comment by deronnax** [comment]
Yesterday I had an error message from pyproject-fmt complaining that it did not find tox in the path. I did not get why pyproject-fmt was complaining about a missing tox and I thought my python installation was broken so I spent dozen of minutes checking stuff and after a while, I learnt that pyproject-fmt needed tox in the path to generate its classifiers.
It makes double-down that this behavior is very specific to some use-case and should be able to disabled, if not disabled by default and expressively asked for.
Most python non-open source applications will not have tox in the path, because they target only one python version.

**Comment by gaborbernat** [comment]
PR for this would be welcome 👍

**Comment by deronnax** [comment]
please assign this issue to me :)

**Comment by gaborbernat** [comment]
@deronnax still planning to do this?

**Comment by henryiii** [comment]
I’d recommend a `—-max-supported-python` flag (or similar name) also supporting auto (current behavior, default) and none, which means the classifiers are not touched.

**Comment by gaborbernat** [comment]
> I’d recommend a `—-max-supported-python` flag (or similar name) also supporting auto (current behavior, default) and none, which means the classifiers are not touched.

I'm not a fan of this, I'd rather have an off flag that would remove version classifiers if present at all.

**Comment by deronnax** [comment]
@gaborbernat I plan to do it whenever I find the time, but if someone want to do it before me, be my guest.

**Comment by henryiii** [comment]
But how do I control the classifier range? I tried pyproject-fmt on a project, and it's stripping out the 3.13 classifier when I add it. For setup-cfg-fmt, there was a flag you could set to control the maximum classifier. I don't want it stripping classifiers I support, or adding classifiers I don't support. How do I control the upper limit without a setting?

**Comment by gaborbernat** [comment]
https://pyproject-fmt.readthedocs.io/en/latest/#configuration-via-file should be documented there

**Comment by henryiii** [comment]
I didn't know about that, perfect, thanks! This is better than a CLI flag.

**Comment by gaborbernat** [comment]
![Screenshot_20240522-003348.png](https://github.com/tox-dev/pyproject-fmt/assets/690238/c4928804-1391-4e20-b16c-e013f26c3fc4)

🤔 Maybe this should be in bold characters.

**Comment by henryiii** [comment]
I think might be just because I am familiar with the project from before it had this option. :) This was very different last I read it: https://pyproject-fmt.readthedocs.io/en/latest/#python-version-classifiers - it discussed tox then.

**Comment by gaborbernat** [comment]
Yeah just got reworked with version 2, we no longer use tox to calculate the max version.
